source/core/slang-performance-profiler.cpp

```diff
@@ -65,7 +65,7 @@ namespace Slang
 
             if (strSize > 0)
             {
-                strncpy_s(profileEntry.funcName, strSize, func.key, strSize);
+                memcpy(profileEntry.funcName, func.key, strSize);
```

Comment:
It looks like you will be missing the null terminator with the new code.
As an example, when strlen returns a value of 1. You need copy two characters not one because you will need to include the null terminator.
Isn't it?

Reply:
I don't think It will miss the null terminator.
Because:

1. the funcName is initialize to all 0
2. the strSize is min(sizeof(funcName)-1, strlen(func,key)), so it means that I will copy at most `sizeof(funcName)-1` characters, so that will always make funcName has null terminator at end.

Reply:
Probably we should not initialize the member to zero and copy the null terminator explicitly.
It is easy to fix by adding 1 like following,
```
memcpy(profileEntry.funcName, func.key, strSize + 1);
```

Reply:
Actually I take this back.
The last character might not be a null terminator if it was longer than 255.
It should be more like
```
profileEntry.funcName[strSize] = 0;
```